src/lib/storage/storage_manager.hpp

```diff
@@ -11,16 +12,42 @@
 
 #include "lqp_view.hpp"
 #include "prepared_plan.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/dictionary_segment.hpp"
 #include "types.hpp"
 
 namespace hyrise {
 
 class Table;
 class AbstractLQPNode;
 
+const auto MAX_CHUNK_COUNT_PER_FILE = uint8_t{50};
+
+struct FILE_HEADER {
+  uint32_t storage_format_version_id;
+  uint32_t chunk_count;
+  std::array<uint32_t, MAX_CHUNK_COUNT_PER_FILE> chunk_ids;
+  std::array<uint32_t, MAX_CHUNK_COUNT_PER_FILE> chunk_offset_ends;
+};
+
+struct CHUNK_HEADER {
+  uint32_t row_count;
+  std::vector<uint32_t> segment_offset_ends;
+};
+
+enum class PersistedSegmentEncodingType : uint32_t {
+  Unencoded,
+  DictionaryEncoding8Bit,
+  DictionaryEncoding16Bit,
+  DictionaryEncoding32Bit,
+  DictionaryEncodingBitPacking
+};
+
 // The StorageManager is a class that maintains all tables
 // by mapping table names to table instances.
```

Comment:
We should adapt this comment as soon as we are sure about the function of the StorageManager in our implementation.